src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -229,4 +232,83 @@ module.exports = class LevelPlane extends Array {
 
     return this[myIndex].blockType;
   }
+
+  getRedstone() {
+    this.redstoneList = [];
+    this.redstoneListON = [];
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType.substring(0,12) === "redstoneWire") {
+        this[i].isPowered = false;
+        let position = [];
+        if (this.levelModel !== null) {
+          position = this.levelModel.controller.levelView.indexToCoordinates(i);
+        }
+        this.redstoneList.push(position);
+      }
+    }
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType === "railsRedstoneTorch") {
+        let position = [];
+        if (this.levelModel !== null) {
+          position = this.levelModel.controller.levelView.indexToCoordinates(i);
+        }
+        this.redstonePropagation(position);
+      }
+    }
+
+    for (let obj in this.redstoneList) {
+      if (this.levelModel !== null) {
```

Comment:
Once you move `coordinatesToIndex` I don't think this null check is needed anymore?